db/migrate/20200911142816_add_deleted_at_to_teams.rb

```diff
@@ -0,0 +1,8 @@
+class AddDeletedAtToTeams < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
+
+  def change
+    add_column :teams, :deleted_at, :datetime
+    add_index :teams, :deleted_at, algorithm: :concurrently
```

Comment:
I wonder if a partial index would be helpful here, as we'd normally be querying for non-deleted teams?

```suggestion
    add_index :teams, :deleted_at, name: 'not_deleted_index', where: {deleted_at: nil), algorithm: :concurrently
```

(Probably marginal gains at our scale though)